src/anomalib/trainer/loops/predict.py

```diff
@@ -0,0 +1,90 @@
+"""Predict loop."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any
+
+import torch
+from lightning_fabric.utilities import move_data_to_device
+from pytorch_lightning.loops.dataloader.prediction_loop import PredictionLoop
+from pytorch_lightning.loops.epoch.prediction_epoch_loop import PredictionEpochLoop
+
+import anomalib.trainer as core
```

Comment:
Since I need to refer to the trainer here and predict is called inside trainer, it leads to circular import. I took this inspiration from pytorch lightning. They use `import pytorch_lightning as pl` which does the same thing and circumvents the circular import issue. But maybe we can name it something different.